### PR TITLE
Pass token in individual calls, remove default location for client token

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
A couple of points addressed here:

1. If you want to use Navigator against a ledger with authentication, you _have_ to explicitly provide a path where the token is stored and something has to be found, otherwise the process will crash before it fully starts (this prevents involuntarily sending your token to the wrong ledger)
2. the `LedgerClient` now asks to set the token explicitly for every call, enabling a more fine grained handling from the caller

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by dasormeter** [comment]
hiya @stefanobaghino-da in its default mode from a vanilla SDK install sandbox is essentially in open world mode, right?  it has authentication set to provide all privileges to all users and does not do token validation, is that right?  or will sandbox soon require that you need to pass a token for it to work in default configuration?

**Comment by stefanobaghino-da** [comment]
> hiya @stefanobaghino-da in its default mode from a vanilla SDK install sandbox is essentially in open world mode, right? it has authentication set to provide all privileges to all users and does not do token validation, is that right? or will sandbox soon require that you need to pass a token for it to work in default configuration?

Be aware that when talking about "open world" we usually refer to using parties without allocation (which is indeed the case for the sandbox).

In terms of authentication, the sandbox runs by default with a pass-through implementation of authentication that doesn't check headers and automatically authorizes any call. This mode will always be supported for development, even though specific implementation details may vary, as we recently agreed to have the token being able to optionally carry meaningful information that will be used to pre-fill your request (e.g. ledger identifier). But again, the default behavior for dev is going to be "god mode".

If you want to run the sandbox in production it's necessary to start it specifying an option that enables authorization. @rautenrieth-da merged a PR today (#3283) that introduced JWT with an hard-coded secret and he's currently working on another contribution that will enable using JWT using a key pair.

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/LedgerClient.scala [hunk 1]
```diff
@@ -106,36 +125,18 @@ object LedgerClient {
...
+      esf: ExecutionSequencerFactory): Future[LedgerClient] =
     apply(
-      LedgerIdentityServiceGrpc.stub(channel).withCallCredentials(creds),
-      TransactionServiceGrpc.stub(channel).withCallCredentials(creds),
-      ActiveContractsServiceGrpc.stub(channel).withCallCredentials(creds),
-      CommandSubmissionServiceGrpc.stub(channel).withCallCredentials(creds),
-      CommandCompletionServiceGrpc.stub(channel).withCallCredentials(creds),
-      CommandServiceGrpc.stub(channel).withCallCredentials(creds),
-      PackageServiceGrpc.stub(channel).withCallCredentials(creds),
+      LedgerIdentityServiceGrpc.stub(channel),
```
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/LedgerClient.scala]: It's not necessary to have it, you can simply build a `Channel`, add credentials to it and pass it in. That's how I made `AuthorizationIT` work without appending credentials to each call. :slightly_smiling_face: The next contribution in this Yak shaving sequence will make the approach a bit more idiomatic using the client (by allowing to pass in a pre-configured builder).

[the remark this answers is not in the document]
Comment (gerolf-da) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/LedgerClient.scala]: But there the `CallCredentials` are also not attached to the channel, but to the stub, recreating the `LedgerContext` that now contains stubs initialized with those call credentials.

**Pull request merged**